doc/installation.rst

```diff
@@ -257,54 +314,76 @@ You'll first want to create and activate a virtual environment in which to insta
 
 With ``virtualenvwrapper``:
 
-.. code-block:: bash
+.. code-block:: shell
 
    mkvirtualenv spyder-env
    workon spyder-env
 
 Otherwise, on macOS/Linux/Unix:
 
-.. code-block:: bash
+.. code-block:: shell
 
    python3 -m venv spyder-env
    source spyder-env/bin/activate
 
 or on Windows:
 
-.. code-block:: bash
+.. code-block:: batch
 
    python -m venv spyder-env
-   spyder-env/Scripts/activate.bat
+   spyder-env\Scripts\activate.bat
 
-After activating your environment, to install Spyder and its other dependencies, run ``pip install spyder``.
+After activating your environment, to install Spyder and its optional dependencies, run:
+
+.. code-block:: shell
+
+   pip install spyder numpy scipy pandas matplotlib sympy cython
+
+Or for a minimal installation, run:
+
+.. code-block:: shell
+
+   pip install spyder
 
 .. image:: /images/installation/installation-pip-install.gif
    :alt: Running Spyder installation with pip
 
-You may need to install a Qt binding (PyQt5) separately with ``pip`` if running under Python 2.
+To launch Spyder after installing, ensure your environment is activated and run the ``spyder`` or ``spyder3`` command.
 
-To launch Spyder after installing, ensure your environment is activated and run the ``spyder3`` command.
 
 
+.. _install-updating:
 
 ===============
 Updating Spyder
 ===============
 
-To update Spyder installed via our :ref:`standalone packages<standalone_installers_ref>` on Windows and macOS, you'll currently need to manually download and install the latest release (if on Windows, make sure to remove the old version first from Control Panel/System Settings).
+By default, Spyder checks for updates automatically on startup.
+You can also do manually with :menuselection:`Help --> Check for updates` on Windows and Linux, or :menuselection:`Spyder --> Check for updates` on macOS.
+
+The :ref:`install-standalone` for Spyder 5.4.0+ also include update functionality built right into Spyder, which after checking for updates as above will display a prompt to automatically download and install the current version.
+On earlier versions, you'll need to manually download and install the latest release (if on Windows, make sure to remove the old version first from Control Panel/System Settings).
 
-If you installed Spyder through Anaconda, WinPython, MacPorts, or your system package manager, update using those same methods.
-With Anaconda, just run (in Anaconda Prompt if on Windows, otherwise in your system terminal):
+If you installed Spyder through Conda, WinPython, Homebrew, MacPorts, or your system package manager, update using those same methods, as described in their respective documentation.
+With Conda-based installs, just run (in Anaconda Prompt if on Windows, otherwise in your system terminal):
 
-.. code-block:: bash
+.. code-block:: shell
 
-   conda update anaconda
```

Comment:
Most of the time users get a package resolution conflicts without running `conda update anaconda` first. So, we should keep this command.

Furthermore, these commands could also fail sometimes with the same kind of error. In that case, they've told us they need to run

    conda remove spyder
    conda install spyder

Reply:
Well, that's only true of Anaconda, whereas `conda update anaconda` will generally fail with an error on the other three, as the `anaconda` metapackage will generally not be installed. And are you sure that workaround works reliably? Wouldn't it be better to just tell users to re-solve/re-create their environment (and probably faster too, because it avoids two resolution steps and a clean solve is much faster than an incremental one)?

Reply:
> Well, that's only true of Anaconda

That's right. Then we'd need to differentiate at this point between updating Spyder in Anaconda and Miniconda, Miniforge, etc.

> And are you sure that workaround works reliably?

Yes, and it's also important to mention the second workaround I posted above.

> Wouldn't it be better to just tell users to re-solve/re-create their environment (and probably faster too, because it avoids two resolution steps and a clean solve is much faster than an incremental one)?

That's not a possibility for `base` and the problem with installing Spyder in other envs is that you have to connect it to `base` and keep track that the right version of Spyder-kernels is installed on it.

Reply:
Thanks:

* I thoroughly revised the Updating section to incorporate these improvements and others.
* It then was long enough that I re-organized it and split it into subsections per method to be easier to follow.
* It it turn occurred to me that it would make more sense just putting it under the respective top-level method headings, like the "Running Spyder" sections are, so everything for that method is in one place.
* That then motivated me to re-organize the Conda section into separate subsections for Conda envs and base Anaconda, and to put the former before the latter to further emphasize it.
* Finally, I also moved the "Alternative Methods" section below the pip section, as unlike pip they are third party and we don't control them at all, so better to make more clear that fact.